nautobot/core/views/mixins.py

```diff
@@ -473,7 +474,11 @@ def get_queryset(self):
         Override the original `get_queryset()` to apply permission specific to the user and action.
         """
         queryset = super().get_queryset()
-        return queryset.restrict(self.request.user, PERMISSIONS_ACTION_MAP[self.action])
+        return queryset.restrict(self.request.user, self.get_action())
+
+    def get_action(self):
+        """Helper method for retrieving action and if action not set defaulting to action name."""
+        return PERMISSIONS_ACTION_MAP.get(self.action, self.action)
 
```

Comment:
Can we add a custom action to the example_plugin and a unit test to ensure it works as expected w/ regard to permissions enforcement?

Reply:
Will do